integration-tests/rpc_integration_test.go

```diff
@@ -0,0 +1,465 @@
+	require.NoError(t, err)
+	_, err = waitForLogSubstring(ctx, logMessages2, "failed to connect to bootstrap peer")
+	require.NoError(t, err)
+
+	client, err := rpc.NewClient(standaloneRPCEndpoint + strconv.Itoa(rpcPort+count1))
+	require.NoError(t, err)
+
+	require.NoError(t, client.AddPeer(expectedPeerInfo))
+
+	log, err = waitForLogSubstring(ctx, logMessages1, "found peer who speaks our protocol")
+	require.NoError(t, err, "didn't find peer")
+
+	var foundPeerLog struct {
+		PeerId   string `json:"remotePeerID_string"`
+		Protocol string `json:"protocol_string"`
+	}
+
+	err = json.Unmarshal([]byte(log), &foundPeerLog)
+	require.NoError(t, err)
+
+	parsedFoundPeerID2, err := peer.IDB58Decode(foundPeerLog.PeerId)
+	require.NoError(t, err)
+	assert.Equal(t, parsedFoundPeerID2, parsedPeerID2)
+	assert.Equal(t, foundPeerLog.Protocol, protocolString)
+
+	log, err = waitForLogSubstring(ctx, logMessages2, "found peer who speaks our protocol")
+	require.NoError(t, err, "didn't find peer")
+
+	err = json.Unmarshal([]byte(log), &foundPeerLog)
+	require.NoError(t, err)
+
+	parsedFoundPeerID1, err := peer.IDB58Decode(foundPeerLog.PeerId)
+	require.NoError(t, err)
+	assert.Equal(t, parsedFoundPeerID1, parsedPeerID1)
+	assert.Equal(t, foundPeerLog.Protocol, protocolString)
+
+	log, err = waitForLogSubstring(ctx, logMessages1, "found peer who speaks our protocol")
+	require.NoError(t, err, "didn't find peer")
+
+	err = json.Unmarshal([]byte(log), &foundPeerLog)
+	require.NoError(t, err)
+
+	parsedFoundPeerID2, err = peer.IDB58Decode(foundPeerLog.PeerId)
+	require.NoError(t, err)
+	assert.Equal(t, parsedFoundPeerID2, parsedPeerID2)
+	assert.Equal(t, foundPeerLog.Protocol, protocolString)
+
+	log, err = waitForLogSubstring(ctx, logMessages2, "found peer who speaks our protocol")
+	require.NoError(t, err, "didn't find peer")
+
+	err = json.Unmarshal([]byte(log), &foundPeerLog)
+	require.NoError(t, err)
+
+	parsedFoundPeerID1, err = peer.IDB58Decode(foundPeerLog.PeerId)
+	require.NoError(t, err)
+	assert.Equal(t, parsedFoundPeerID1, parsedPeerID1)
+	assert.Equal(t, foundPeerLog.Protocol, protocolString)
+
+	// Cancel the context and wait for all outstanding goroutines to finish.
+	cancel()
+	wg.Wait()
+}
+
+func TestGetStats(t *testing.T) {
+	teardownSubTest := setupSubTest(t)
+	defer teardownSubTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Remove the old database and p2p files.
+	removeOldFiles(t, ctx)
+
+	buildStandaloneForTests(t, ctx)
+
+	logMessages := make(chan string, 1024)
+
+	count := int(atomic.AddInt32(nodeCount, 1))
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		startStandaloneNode(t, ctx, count, logMessages)
+	}()
+
+	log, err := waitForLogSubstring(ctx, logMessages, "started RPC server")
+	require.NoError(t, err, "RPC server didn't start")
+	var jsonLog map[string]interface{}
+	err = json.Unmarshal([]byte(log), &jsonLog)
```

Comment:
You are committing the private key for each standalone node in this PR. If we're doing that, then we don't need to extract the peer ID from the logs. We should either:

1. Don't commit the private key for each standalone node in this PR.
2. Extract the expected peer ID directly from the private key instead of reading the logs.